dts/arm/nordic/nrf51822.dtsi

```diff
@@ -24,6 +24,7 @@
 		ficr: ficr@10000000 {
 			compatible = "nordic,nrf-ficr";
 			reg = <0x10000000 0x1000>;
+			#nordic,ficr-cells = <1>;
```

Comment:
note: I've opened https://github.com/zephyrproject-rtos/zephyr/pull/68331 so we can simplify this PR a bit and focus on 54H only changes.